src/meta/src/hummock/compaction/overlap_strategy.rs

```diff
@@ -157,13 +155,47 @@ fn check_table_overlap(key_range: &KeyRange, table: &SstableInfo) -> bool {
     key_range.full_key_overlap(&other)
 }
 
+fn check_key_vnode_overlap(info: &SstableInfo, table: &SstableInfo) -> bool {
+    let text = KeyRange::from(info.key_range.as_ref().unwrap());
+    let other = KeyRange::from(table.key_range.as_ref().unwrap());
+    if !text.full_key_overlap(&other) {
+        return false;
+    }
+    let text_len = info.get_vnode_bitmaps().len();
+    let other_len = table.get_vnode_bitmaps().len();
+    let (mut i, mut j) = (0, 0);
+    while i < text_len && j < other_len {
+        let x = &info.get_vnode_bitmaps()[i];
+        let y = &table.get_vnode_bitmaps()[j];
+        match x.get_table_id().cmp(&y.get_table_id()) {
+            Ordering::Less => {
+                i += 1;
+            }
+            Ordering::Greater => {
+                j += 1;
+            }
+            Ordering::Equal => {
+                let maplen = x.get_maplen();
+                assert_eq!(maplen, y.get_maplen());
```

Comment:
Why not just use `get_bitmap().len()` as `maplen`? so `maplen` can be omitted from proto.

Reply:
> Do we really need a field to indicate the length of a bitmap? After all, the number of vnodes is fixed (2048), so that all bitmaps will have the same length, right?

for the sake of completeness

Reply:
> get_bitmap().len()

`get_bitmap().len()` is number of bitmaps rather than maplen

Reply:
> > get_bitmap().len()
> 
> `get_bitmap().len()` is number of bitmaps rather than maplen

But here `bitmap_idx` is used as an index of `Vec<u8>` ?

```
for bitmap_idx in 0..maplen as usize {
                    if (x.get_bitmap()[bitmap_idx] & y.get_bitmap()[bitmap_idx]) != 0 {
```

Reply:
> > > get_bitmap().len()
> > 
> > 
> > `get_bitmap().len()` is number of bitmaps rather than maplen
> 
> But here `bitmap_idx` is used as an index of `Vec<u8>` ?
> 
> ```rust
> for bitmap_idx in 0..maplen as usize {
>     if (x.get_bitmap()[bitmap_idx] & y.get_bitmap()[bitmap_idx]) != 0 {
> ```

will fix

Reply:
fixed